src/main/java/io/appform/dropwizard/actors/base/UnmanagedPublisher.java

```diff
@@ -106,6 +114,83 @@ public final long pendingSidelineMessagesCount() {
         return Long.MAX_VALUE;
     }
 
+
+    public final boolean publishWithConfirmation(Message message, AMQP.BasicProperties properties, long timeout)
+            throws Exception {
+        publishChannel.confirmSelect();
+        final CountDownLatch publishAckLatch = new CountDownLatch(1);
+        publishChannel.addConfirmListener(new ConfirmListener() {
+            public void handleNack(long deliveryTag, boolean multiple){
+                publishAckLatch.countDown();
+                System.out.println("Message : " + message + " not acknowledged");
+            }
+
+            public void handleAck(long deliveryTag, boolean multiple){
+                publishAckLatch.countDown();
+                System.out.println("Message : " + message + " acknowledged");
+            }
+        });
+
+        String routingKey;
+        if (config.isSharded()) {
+            routingKey = NamingUtils.getShardedQueueName(queueName, getShardId());
+        } else {
+            routingKey = queueName;
+        }
+        publishChannel.basicPublish(config.getExchange(), routingKey, properties, mapper().writeValueAsBytes(message));
+
+        if (!publishAckLatch.await(timeout, TimeUnit.MILLISECONDS)) {
+            System.out.println("Timed out waiting for publish acks, Message : " + message);
+            return false;
+        }
+        return true;
+    }
+
+    public final List<Message> publishWithConfirmation(List<Message> messages,  AMQP.BasicProperties properties, long timeout) throws Exception {
+        publishChannel.confirmSelect();
+        ConcurrentNavigableMap<Long, Message> outstandingConfirms = new ConcurrentSkipListMap<>();
+        final CountDownLatch publishAckLatch = new CountDownLatch(messages.size());
+        publishChannel.addConfirmListener(new ConfirmListener() {
+            public void handleNack(long deliveryTag, boolean multiple) {
+                publishAckLatch.countDown();
+            }
+
+            public void handleAck(long deliveryTag, boolean multiple) {
+                if (multiple) {
+                    ConcurrentNavigableMap<Long, Message> confirmed = outstandingConfirms.headMap(
+                            deliveryTag, true
+                    );
+                    confirmed.clear();
+                } else {
+                    outstandingConfirms.remove(deliveryTag);
+                }
+                publishAckLatch.countDown();
+            }
+        });
+
+        String routingKey;
+        if (config.isSharded()) {
+            routingKey = NamingUtils.getShardedQueueName(queueName, getShardId());
+        } else {
+            routingKey = queueName;
+        }
+        for (Message message : messages) {
+            outstandingConfirms.put(publishChannel.getNextPublishSeqNo(), message);
+            publishChannel.basicPublish(config.getExchange(), routingKey, properties,
```

Comment:
in case of exception here what should be the behaviour, also lets log in case of exception

Reply:
in Case of exception, message with be treated as nacked and will be returned by function as List of nacked message.